### PR TITLE
DSND-525 Invoke Company Metrics /recalculate POST endpoint

### PR DESCRIPTION
**Pull request by louiebrown2**

### CODE REVIEW

File: src/main/java/uk/gov/companieshouse/company/metrics/processor/ResponseHandler.java [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+import org.springframework.http.HttpStatus;
+import uk.gov.companieshouse.company.metrics.exception.NonRetryableException;
+import uk.gov.companieshouse.company.metrics.exception.RetryableException;
+import uk.gov.companieshouse.logging.Logger;
+
+import java.util.Map;
+
+public final class ResponseHandler {
+
+    private ResponseHandler() {
```
Comment (ipg-ch) [inline src/main/java/uk/gov/companieshouse/company/metrics/processor/ResponseHandler.java:12]: Private function that does nothing?

File: src/main/java/uk/gov/companieshouse/company/metrics/service/api/BaseApiClientServiceImpl.java [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+import org.springframework.web.server.ResponseStatusException;
+import uk.gov.companieshouse.api.error.ApiErrorResponseException;
+import uk.gov.companieshouse.api.handler.Executor;
+import uk.gov.companieshouse.api.handler.exception.URIValidationException;
+import uk.gov.companieshouse.api.model.ApiResponse;
+import uk.gov.companieshouse.logging.Logger;
+
+
+public abstract class BaseApiClientServiceImpl {
+    protected Logger logger;
```
Comment (ipg-ch) [inline src/main/java/uk/gov/companieshouse/company/metrics/service/api/BaseApiClientServiceImpl.java:15]: Should this not be final to protect from being changed by inheriting classes?

File: src/test/java/uk/gov/companieshouse/company/metrics/service/CompanyMetricsApiServiceTest.java [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+        final ApiResponse<Void> expected = new ApiResponse<>(
+                HttpStatus.OK.value(), Collections.emptyMap());
+
+        when(privateCompanyMetricsUpsertHandler.postCompanyMetrics(MOCK_COMPANY_METRICS_RECALCULATE_URI, metricsRecalculateApi)).thenReturn(privateCompanyMetricsUpsert);
+        when(privateCompanyMetricsUpsert.execute()).thenReturn(expected);
+
+        final ApiResponse<Void> response = companyMetricsApiService.postCompanyMetrics(
+                MOCK_CONTEXT_ID, MOCK_COMPANY_NUMBER, metricsRecalculateApi);
+
+        assertThat(response).isSameAs(expected);
```
Comment (ipg-ch) [inline src/test/java/uk/gov/companieshouse/company/metrics/service/CompanyMetricsApiServiceTest.java:84]: Not sure about this but isSameAs jus do reference comparison? If so contents could have changes and test will pass. stronger verification would be to test that the status code and header are as expected.

**Comment by louiebrown2** [comment]
Comments and suggested changes discussed over huddle

File: src/test/java/uk/gov/companieshouse/company/metrics/processor/CompanyMetricsProcessorTest.java [hunk 4]
```diff
@@ -32,21 +30,27 @@ public class CompanyMetricsProcessorTest {
...
     }
 
     @Test
     @DisplayName("Successfully processes a kafka message containing a ResourceChangedData payload and extract companyNumber")
     void When_ValidResourceChangedDataMessage_Expect_MessageProcessedAndExtractedCompanyNumber() throws IOException {
         Message<ResourceChangedData> resourceChangedDataMessage = testData.createResourceChangedMessage();
-        companyMetricsProcessor.process(resourceChangedDataMessage);
+        companyMetricsProcessor.process(resourceChangedDataMessage, topic, partition, offset);
 
         verify(logger, atLeastOnce()).trace((
```
[the remark this answers is not in the document]
Comment (louiebrown2) [reply src/test/java/uk/gov/companieshouse/company/metrics/processor/CompanyMetricsProcessorTest.java:55]: Consider for tech debt

**Review by ipg-ch — APPROVED** [review]

**Pull request merged**